### PR TITLE
Supporting Vault Mount Path

### PR DESCRIPTION
**Pull request by Vishesh-Gupta**
- Add support for a different mount_path

### CODE REVIEW

File: tests/test_gestalt.py [hunk 1]
```diff
@@ -471,8 +470,27 @@ def test_vault_get():
...
+def test_vault_mount_path():
+    g = gestalt.Gestalt()
+    g.build_config()
+    client_config = gestalt.HVAC_ClientConfig()
+    client_config['url'] = ""
+    client_config['token'] = "myroot"
+    client_config['cert'] = None
+    client_config['verify'] = True
+    g.add_vault_config_provider(client_config, auth_config=None)
+    print("Requires the user to set a token in the client")
```
Comment (john-delivuk) [inline tests/test_gestalt.py:490]: Should we be printing?

File: tests/test_gestalt.py [hunk 2]
```diff
@@ -471,8 +470,27 @@ def test_vault_get():
...
+    assert secret == client_password
+
+
+def test_vault_mount_path():
+    g = gestalt.Gestalt()
+    g.build_config()
+    client_config = gestalt.HVAC_ClientConfig()
+    client_config['url'] = ""
+    client_config['token'] = "myroot"
+    client_config['cert'] = None
```
Comment (john-delivuk) [inline tests/test_gestalt.py:487]: Are default values set here and we're overriding them?

File: tests/test_gestalt.py [hunk 3]
```diff
@@ -471,8 +470,27 @@ def test_vault_get():
...
+    client_config['verify'] = True
+    g.add_vault_config_provider(client_config, auth_config=None)
+    print("Requires the user to set a token in the client")
+    client_id_mount_path = "test_mount"
```
Comment (john-delivuk) [inline tests/test_gestalt.py:491]: might want to make this `test_path` to avoid confusion for the next person

**Review by john-delivuk — COMMENTED** [review]
Minor remarks, overall looks good.

File: tests/test_gestalt.py (hunk 1 above)
Comment (Vishesh-Gupta) [reply tests/test_gestalt.py:490]: Yes! The reason why I want to print this is an assurance for the test writer that the token is manually added in the test action

File: tests/test_gestalt.py (hunk 2 above)
Comment (Vishesh-Gupta) [reply tests/test_gestalt.py:487]: Yes :)

File: tests/test_gestalt.py (hunk 3 above)
Comment (Vishesh-Gupta) [reply tests/test_gestalt.py:491]: this is just a value, which is why `client_id_mount_path` is used everywhere. We don't work with `test_mount` furthermore than this step. :)

**Review by john-delivuk — APPROVED** [review]

**Review by tanishq-dubey — APPROVED** [review]

**Pull request merged**